### ISSUE TITLE
no scrolling on Hacker News or Wikipedia

### ISSUE THREAD

**Issue by nickpsecurity** [servo/servo#12082]
I loaded the comments for today's thread on HN for Servo. Couldn't see past the first replies since Page Down and Up did nothing. No scrollbar either. Tested to see if it was HN-specific by going to Wikipedia, loading up a long article. Same effect. On Ubuntu 16.04 bare-metal loading Servo Linux prepackaged version w/ -c option on startup for non-GPU rendering & no non-standard functionality in this Ubuntu build.

**Comment by notriddle** [comment]
We don't have a scrollbar or proper handling of keyboard navigation. We do have support for a mouse wheel, if your computer has that. Laptops with trackpads usually have a scroll-area on the side that acts like a mouse wheel, or they act as a mouse wheel when you move two fingers across them at once.

**Comment by nickpsecurity** [comment]
"Laptops with trackpads usually have a scroll-area on the side that acts like a mouse wheel"

I didn't even know that after all these years. Thought the drawings were just suggesting how to use the trackpad. Well, tried that and it doesn't work on Firefox or Servo. So, maybe I don't have that feature. Tested with mouse scrolling with USB mouse. Scrollwheel on HN works on Firefox but not on Servo. Confirmed same on Wikipedia.

Note: While we're at it, is right-click a left-click equivalent right now?

**Comment by jdm** [comment]
Yeah, we don't differentiate between mouse buttons at the moment.

**Comment by nickpsecurity** [comment]
Update. Rebooted my computer as it was getting finicky. My touchpad *does* scroll pages up and down as I confirmed in Firefox on Hacker News and Wikipedia. Must've been driver error. However, it *doesn't* do that on Servo. I dragged the window around, clicked some text, and so on to confirm it had focus. Went back to Firefox, still scrolled, back to Servo, still doesn't. So, might be a legit bug in Servo despite my HW issue.
